### PR TITLE
Check that there is a non-empty narrative when there is no ref

### LINKED ISSUE #50: Add rules (regex matches): ref attributes
> If this is not present then the narrative MUST contain the name of the organisation (read: there is content). - REGEX

**Impacted attributes:**

- [x] `iati-activities/iati-activity/planned-disbursement/provider-org/@ref`
- [x] `iati-activities/iati-activity/planned-disbursement/receiver-org/@ref`
- [x] `iati-activities/iati-activity/transaction/provider-org/@ref`
- [x] `iati-activities/iati-activity/transaction/receiver-org/@ref`
- [x] `iati-organisations/iati-organisation/recipient-org-budget/recipient-org/@ref`

Regex for non-space character: `^(?!\s*$).+` ([source](https://stackoverflow.com/a/3085553))

Logic:

```
if @ref is not present:  # condition
    narrative.text() matches ^(?!\s*$).+
```

### PR DESCRIPTION
**Pull request by hayfield**
Fixes #50

### CODE REVIEW

File: rulesets/standard.json [hunk 1]
```diff
@@ -57,6 +57,14 @@
             ]
         }
     },
+    "//planned-disbursement/provider-org | //planned-disbursement/receiver-org | //transaction/provider-org | //transaction/receiver-org | //recipient-org-budget/recipient-org": {
+        "regex_matches": {
+            "cases": [
+                { "regex": "^(?!\s*$).+",
+                  "paths": [ "narrative" ], "condition": "count(@ref)==1" }
```
Comment (dalepotter) [inline rulesets/standard.json]: Wonder if we should look for `narrative/text()` here?

Comment (hayfield) [reply rulesets/standard.json]: Because it's an element, this is automatically appended.

Comment (hayfield) [reply rulesets/standard.json]: https://github.com/IATI/IATI-Rulesets/blob/version-2.02/iatirulesets/__init__.py#L11

Comment (dalepotter) [reply rulesets/standard.json]: Sounds good - as we found, this doesn't appear in the [ruleset spec](http://iatistandard.org/202/rulesets/ruleset-spec) so would be oblivious to a ruleset user.  I'll make a Monday project to try to improve the writeup of the spec at some point!

**Pull request merged**